File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
import java.math.BigInteger;

class Solution {
    public BigInteger solution(int balls, int share) {
        BigInteger num1 = new BigInteger("1");
        BigInteger num2 = new BigInteger("1");

        for (int i = (balls - share + 1); i <= balls; i++) {
            num1 = num1.multiply(BigInteger.valueOf(i));
        }
        for (int i = 1; i <= share; i++) {
            num2 = num2.multiply(BigInteger.valueOf(i));
        }

        return num1.divide(num2);
    }
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.js
function solution(balls, share) {
    let num1 = 1;
    let num2 = 1;
    
    if (share === balls) {
        return 1;
    } else if((balls - share) > share) {
        for (let i = 1; i <= share; i++) {
            num1 *= (balls - i + 1);
            num2 *= i;
        }
        
        return num1 / num2;
    } else {
        for (let i = 1; i <= (balls - share); i++) {
            num1 *= (balls - i + 1);
            num2 *= i;
        }
        
        return num1 / num2;
    }
}